Feature#35 dashboard

xFerchoVJ opened the pull request:
# Description
What did you implemented/Why did you implemented this:
 - I created the dashboard and made it functional

 Example:
  - I created the dashboard and also integrated the routes for management.
  
  ## Screenshots 
![image](https://user-images.githubusercontent.com/76576025/126837065-32755d96-5185-4626-820e-b906cfaf3f48.png)

xFerchoVJ committed:
Create folders

I've created the folders for companies and employees

To make functional the dashboard i created those folders
and i also created the index page for both folders

xFerchoVJ committed:
Create admin.scss file

I've created this file to create the styles for the dashboard

xFerchoVJ committed:
Add routes

Add resources for employees, companies and admin

xFerchoVJ committed:
Dashboard page

I've started making the dashboard to the admin be able manage the employees, the companies and the reports

xFerchoVJ committed:
Create partial

I've created a partial to show it on the dashboard

xFerchoVJ committed:
Dashboard finished

I finished creating the dashboard for the administrator.

xFerchoVJ committed:
Change file

I've change the sign out action for the admin

It was necessary to add the delete method in the action.

YilKanda commented on checador/app/views/layouts/application.html.erb:
```diff
@@ -5,15 +5,15 @@
-    <p class="notice"><%= notice %></p>
-    <p class="alert"><%= alert %></p>
-    
+    <!--<p class="notice"><%= notice %></p>
+    <p class="alert"><%= alert %></p>-->
```
If you don't occupy it, delete it.

YilKanda replied:
It is recommended that you do not leave code in comments.

Remember that Git allows us to view this code on GitHub, in the commit in which it was created.

xFerchoVJ replied:
I'm going to use it later, thank you, Yil

YilKanda commented on checador/app/views/employees/index.html.erb:
```diff
@@ -0,0 +1 @@
+<h2>Manage Employees</h2>
```
Add a blank line.

YilKanda commented on checador/app/views/employees/_index.html.erb:
```diff
@@ -0,0 +1 @@
+<h1>All employees</h1>
```
Add a blank line.

YilKanda commented on checador/app/views/companies/index.html.erb:
```diff
@@ -0,0 +1 @@
+<h1>Company Branches</h1>
```
Add a blank line.

YilKanda commented on checador/app/views/admin/dashboard.html.erb:
```diff
@@ -1,9 +1,60 @@
+
+      </div>
+      
+</div>
```
Add a blank line.

YilKanda commented on checador/app/views/admin/dashboard.html.erb:
```diff
@@ -1,9 +1,60 @@
+
+      <div class="container">
+        <div class="d-flex justify-content-between align-items-center p-3">
+              <h2 class="margin">Welcome back, Admin!</h2>
+              <p class="m-0 fst-italic text-muted">
+                See the latest information
+              </p>
```
Check the indentation.

YilKanda commented on checador/app/assets/stylesheets/admin.scss:
```diff
@@ -0,0 +1,45 @@
+.margin{
+  margin-top: 15px;
+  margin-bottom: 0px;
+}
```
Add a blank line.

YilKanda commented on checador/app/assets/stylesheets/admin.scss:
```diff
@@ -0,0 +1,45 @@
+}
+
+#sidebar-container {
+    min-height: 100vh;
+    .logo {
+      padding: 0.875rem 3.25rem;
+    } 
+    .menu {
+        width: 18rem;
+        a{
+          font-size: 17px;
+          text-decoration: none;
+      }
+    }
```
Check indentation

YilKanda commented on checador/app/assets/stylesheets/admin.scss:
```diff
@@ -0,0 +1,45 @@
+
+
+.bg.primary { 
+    background-color: $primary-color important;
```
As mentioned @jaimelr, 

> Avoid the use of !important whenever possible. This makes it hard to reuse styles and can mess up your code later when the codebase grows.

YilKanda commented on checador/app/assets/stylesheets/admin.scss:
```diff
@@ -0,0 +1,45 @@
+    background-color: $primary-color important;
+}
+.bg-light {
+  background-color: white important;
```
As mentioned @jaimelr, 

> Avoid the use of !important whenever possible. This makes it hard to reuse styles and can mess up your code later when the codebase grows.

xFerchoVJ replied:
Okay, thank you, let me change it! :)

YilKanda commented on checador/app/views/admin/dashboard.html.erb:
```diff
@@ -1,9 +1,60 @@
+            </h4>
+        </div>
+        <div class="menu">
+            <a href="/companies" class="d-block p-3 text-light"> 
+                <i class="icon ion-md-business mx-2 lead"></i> 
+                Manage Company Branches 
+            </a>
+            <a href="/employees" class="d-block p-3 text-light"> 
+                <i class="icon ion-md-contacts mx-2 lead"></i>
+                Manage Employees 
+            </a>
+            <a href="#" class="d-block p-3 text-light"> 
+                <i class="icon ion-md-stats mx-2 lead"></i>
+                Reports 
+            </a>
+        </div>
```
Check the indentation.

YilKanda commented on checador/app/views/admin/dashboard.html.erb:
```diff
@@ -1,9 +1,60 @@
+              </a>
+              <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
+                <li>
+                  <!--<a class="dropdown-item" href="#">Action</a>-->
```
If they don't occupy this line, delete it.

YilKanda reviewed (changes requested):
Check the indentation of your files, since they use 2 spaces in some parts and 4 spaces in others. Remember that they must be uniform in the use of spaces (I recommend always using 2).

I also recommend that in the texts, instead of using plain text, they use [Rails Internationalization (I18n)](https://guides.rubyonrails.org/i18n.html), so that they can implement their application, not only in Spanish, but in several other languages ​​(for example, English), and in turn , allows them to have better control of their texts.

xFerchoVJ committed:
Change files

Change the files where i made some errors such as indentation and some blank lines
that i had to add.

JoseRicardoJimenez reviewed (dismissed):
nice!

YilKanda reviewed (commented):
It is recommended that you do not leave code in comments.

Remember that Git allows us to view this code on GitHub, in the commit in which it was created.

xFerchoVJ committed:
Flash partial

I've made this file to be able to use the flashes in any other view

xFerchoVJ committed:
Modify views

I've removed the messages and i moved them to the corresponding view

SEVERO96 committed:
Merge branch 'master' into feature#35_dashboard

RobertoCarMed reviewed (approved):
LGTM!

YilKanda reviewed (approved):
LGTM!
